File: src/Host/ClientApp/src/entities/provinces/ProvinceWithUser.ts
import { apiRequester } from "../../shared/ApiRequester";

export interface ProvinceWithUserState {
    data: ProvinceWithUser,
    isLoading: boolean,
    isChecked: boolean,
    error: string
}

export interface ProvinceWithUser {
    province: Province,
    faction: Faction,
    user: User | undefined,
}

export interface Province {
    id: number
}

export interface Faction {
    id: number,
    name: string,
    gold: number,
    investments: number,
    fortifications: number,
    fortificationCoef: number,
    size: number,
    userId: string,
    suzerainId: number | undefined,
    turnOfDefeat: number
}

export interface User {
    id: number,
    userName: string,
}

const extendedApiSlice = apiRequester.injectEndpoints({
    endpoints: builder => ({

        getProvinceWithUser: builder.query<ProvinceWithUser, number>({
            query: (params) => ({
                url: 'provinces',
                params: {
                    id: params
                }
            }),
            providesTags: [{ type: 'Daily', id: 'ProvinceWithUser' }]
        }),
    })
})

export const { useGetProvinceWithUserQuery } = extendedApiSlice;

File: src/Domain/Provinces/ProvinceWithUser.cs
using YAGO.World.Domain.Factions;
using YAGO.World.Domain.Users;

namespace YAGO.World.Domain.Provinces
{
    public class ProvinceWithUser
    {
        public Province Province { get; }
        public Faction Faction { get; }
        public User? User { get; }

        public ProvinceWithUser(Province province, Faction faction, User user)
        {
            Province = province;
            Faction = faction;
            User = user;
        }
    }
}
